tl-tl/sc/white-paper.md

```diff
@@ -2,67 +2,67 @@
 
-Smart contracts in NeoContract can call each other, but not be recursively called. Recursion can be achieved within the contract, but it cannot cross the boundaries of the current contract. In addition, the call relationship between contracts must be static: The target cannot be specified at runtime. This allows the behavior of the program to be fully determined before execution, and its call relationship to be fully defined before it can run. Based on this, multiple contracts can be dynamically partitioned to achieve parallel execution.
+### 2.2 Mataas na Pagganap
 
-### 2.2 High performance
+Ang environment ng pagpapatupad ng isang matalinong kontrata ay naglalaro ng isang mahalagang papel sa pagganap nito. Kapag ating pinag-aralan ang pagganap sa anumang environment ng pagpapatupad, may dalawang pangunahing tagapagpahiwatig na kritikal:
 
-The execution environment of a smart contract plays an integral role in its performance. When we analyze the performance of any execution environment, there are main two indicators which are critical:
+1. Bilis ng pagpapatupad ng tagubilin
+2. Bilis ng startup ng environment na pagpapatupad
 
-1. Execution speed of the instruction
-2. Startup speed of the execution environment itself
-
-For smart contracts, the execution environment is often more important than the speed of execution of the instruction. Smart contracts are more involved in IO operation of the logic, to determine the instructions, where the implementation of these instructions can easily be optimized. Every time the smart contract is called, it must start up a new virtual machine / container. Therefore, the execution speed of the environment itself (starting a virtual machine / container) has greater impact on the performance of the smart contract system. 
+Para sa matalinong mga kontrata, ang environment ng pagpapatupad ay kadalasang mas importante kaysa sa bilis ng pagpapatupad ng tagubilin. Ang matalinong mga kontrata ay mas kasangkot sa IO na operasyon ng lohika, upang matukoy ang mga tagubilin, kung saan ang implementasyon ng mga tagubiling ito ay madaling mapaunlad. Sa bawat pagkakataon na tinawag nag matalinong kontrata, dapat itong magsimula ng isang panibagong virtual machine / container. Samakatuwid, ang bilis ng pagpapatupad sa environment (pagsisimula ng isang virtual machine / container) ay may mas malaking epekto sa pagganap ng matalinong kontrata na sistema.
```

Comment:
`virtual machine` to `bertuwal na makina`

Reply:
Will change it to `birtwal na makina`.